plugins/Ecommerce/Controller.php

```diff
@@ -42,18 +51,138 @@ public function getSparklines()
 
         $goal = $this->getMetricsForGoal(Piwik::LABEL_ID_GOAL_IS_ECOMMERCE_ORDER);
         foreach ($goal as $name => $value) {
+            if ($name == 'conversion_rate') {
+                $value = $value * 100;
+            }
             $view->$name = $value;
         }
 
         $goal = $this->getMetricsForGoal(Piwik::LABEL_ID_GOAL_IS_ECOMMERCE_CART);
         foreach ($goal as $name => $value) {
+            if ($name == 'conversion_rate') {
+                $value = $value * 100;
+            }
             $name        = 'cart_' . $name;
             $view->$name = $value;
         }
 
         return $view->render();
     }
 
+    /**
+     * Get metrics for an ecommerce goal and add evolution values
+     *
+     * @param      $idGoal
+     * @param null $dataRow
+     *
+     * @return array
+     */
+    protected function getMetricsForGoal($idGoal, $dataRow = null)
+    {
+        $request = new Request("method=Goals.get&format=original&format_metrics=0&idGoal=$idGoal");
+        $datatable = $request->process();
+        $dataRow = $datatable->getFirstRow();
+
+        $return = parent::getMetricsForGoal($idGoal, $dataRow);
+
+        // Previous period data for evolution
+        list($lastPeriodDate, $ignore) = Range::getLastDate();
+        if ($lastPeriodDate !== false) {
+            $date = Common::getRequestVar('date');
+
+            /** @var DataTable $previousData */
+            $previousData = Request::processRequest('Goals.get', ['date' => $lastPeriodDate, 'format_metrics' => 0]);
+            $previousDataRow = $previousData->getFirstRow();
+
+            $return = $this->addSparklineEvolutionValues($return, $idGoal, $date, $lastPeriodDate, $dataRow, $previousDataRow);
+
+        }
+        return $return;
+    }
+
+    /**
+     * Add sparkline evolution figures to the metrics in the supplied array
+     *
+     * @param array         $return
+     * @param string|int    $idGoal
+     * @param string        $date
+     * @param string        $lastPeriodDate
+     * @param DataTable\Row $currentDataRow
+     * @param DataTable\Row $previousDataRow
+     *
+     * @return array
+     */
+    private function addSparklineEvolutionValues(array $return, $idGoal, string $date, string $lastPeriodDate,
+                                                 DataTable\Row $currentDataRow, DataTable\Row $previousDataRow) : array
+    {
+        $metrics = [
```

Comment:
It would be great to get the sparklines standardized everywhere and get rid of the special handling, it will likely cause other issues in the future :+1:

At the start of this change I did spend a bit of time evaluating how much work would be required to convert ecommerce to use standard sparkline visualizations and reports. The layout for ecommerce sparklines is different with three columns instead of two (which don't flow very well) and additional icons. Another issue is that the ecommerce metrics are repurposed goal metrics so they have their labels and descriptions customised. This is likely why special handling was used.

My take was that either the standard sparkline visualizations and macros would have to be enhanced to support more layout configuration options or the ecommerce overview would need to be redesigned into something more consistent with the rest of the application (and possibly new metric classes added to handle the customisation). Both would take some time.

The original issue set an 8hr time limit on the total change and I'm well beyond that after needing to switch sparkline provided data from formatted to unformatted.

Maybe we could add a separate issue to improve the ecommerce sparklines and prioritize it soon after this change so it can benefit from the time spent here?

Reply:
I'm late to the party here but I'd say we should review some of these issues and other refactoring as part of the 5.0.0 milestone coming soon. For now it would be good to get this PR merged as it solves other issues, though we'd need to have some idea of the time involved to best decide here. If it's 1 hour extra to refactor then it's probably worth doing, but if it doubles the time remaining or delays this PR from being merged then I'd agree with @bx80 that we should stick to the original time value of this task as much as possible.